app/modules/assets/models.py

```diff
@@ -145,6 +145,46 @@ def update_symlink(self, asset_asset_group_filepath):
 
         return asset_symlink_filepath
 
+    def mime_type_major(self):
+        if not self.mime_type:
+            return None
+        parts = self.mime_type.split('/')
+        return parts[0]
+
+    def is_mime_type_major(self, major):
+        return self.mime_type_major() == major
+
+    # will only set .meta values that can be derived automatically from file
+    # (will not overwrite any manual/other values); silently fails if unknown type for deriving
+    #
+    #  FIXME - this now is a very basic stub -- it is operating on original file and *very* likely fails
+    #  due to exif/orientation info TODO
+    def set_derived_meta(self):
+        if not self.is_mime_type_major('image'):
+            return None
+        dmeta = {}
+        source_path = self.get_symlink()
+        assert os.path.exists(source_path)
+        with Image.open(source_path) as im:
+            size = im.size
+            dmeta['width'] = size[0]
+            dmeta['height'] = size[1]
+        meta = self.meta if self.meta else {}
```

Comment:
I think this is safe from the issues that @karenc found in the jobs stuff but worth checking if the meta needs a default in this case

Reply:
yup!  will keep an eye on this to make sure its not getting persisted with unexpected values.  thx